pr_agent/git_providers/github_provider.py

```diff
@@ -611,51 +619,35 @@ def remove_reaction(self, issue_comment_id: int, reaction_id: str) -> bool:
     def _parse_pr_url(self, pr_url: str) -> Tuple[str, int]:
         parsed_url = urlparse(pr_url)
 
-        path_parts = parsed_url.path.strip('/').split('/')
-        if self.base_domain in parsed_url.netloc:
-            if len(path_parts) < 5 or path_parts[3] != 'pulls':
-                raise ValueError("The provided URL does not appear to be a GitHub PR URL")
-            repo_name = '/'.join(path_parts[1:3])
+        pattern = r'/repos/([^/]+/[^/]+)/pulls?/(\d+)'
```

Comment:
would this format support

`https://github.com/Codium-ai/pr-agent/pull/1270`

for example ??

Reply:
We would like to support a format such as:
`https://github.enterprise-server.com/api/v3/Codium-ai/pr-agent/pull/1270`

In GitHub Enterprise Server, the API URLs include the `api/v3` segment. The current implementation uses a hard-coded check:

```
if len(path_parts) < 5 or path_parts[3] != 'pulls':
```
This approach does not account for the `api/v3` part of the URL, leading to errors when the URL structure differs. Specifically, this issue affects the following part of the code:

For more information on the URL structure, refer to the GitHub Enterprise Server documentation: 
[GitHub Docs - Get a Pull Request](https://docs.github.com/ja/enterprise-server@3.14/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request)

https://github.com/Codium-ai/pr-agent/blob/ada0a3d10f357e18033b4f6772eac35b79a2c369/pr_agent/servers/github_action_runner.py#L91-L115